dali/operators/generic/one_hot.cc

```diff
@@ -74,13 +78,35 @@ TensorShape<> determine_shape(TensorShape<> in_shape, int num_classes, int axis,
   axis = axis < 0 ? in_shape.sample_dim() : axis;
   int outer_axes = axis;
   int inner_axes = in_shape.sample_dim() - axis;
-  auto outer =  in_shape.first(outer_axes);
+  auto outer = in_shape.first(outer_axes);
   auto inner = in_shape.last(inner_axes);
   return shape_cat(shape_cat(outer, num_classes), inner);
 }
 
 }  // namespace
 
+TensorLayout OneHot::GetOutputLayout(const HostWorkspace &ws, int placement_axis,
+                                     int output_sample_dim) const {
+  const auto &input = ws.template InputRef<CPUBackend>(0);
+  auto in_layout = input.GetLayout();
+  // .size method returns uint_8 which doesn't work well when 0 is printed in error message
+  int in_layout_size = in_layout.size();
+  if (!new_axis_name_) {
+    return {};
+  }
+  // Handles the legacy 'multi-dimensional' scalars-like
+  if (output_sample_dim == 1) {
+    return TensorLayout(&new_axis_name_, 1);
+  }
+  if (in_layout_size + 1 == output_sample_dim) {
+    return in_layout.first(placement_axis) + TensorLayout(&new_axis_name_, 1) +
+           in_layout.last(in_layout_size - placement_axis);
+  }
+  DALI_FAIL(make_string("Input layout mismatch - expected input layout to be of size ",
+                        output_sample_dim - 1, " but instead got \"", in_layout,
+                        "\" that is of size ", in_layout_size, "."));
```

Comment:
```suggestion
                        "\", which is of size ", in_layout_size, "."));
```
Non-defining relative clause always uses "which" with a leading comma.